src/docker/shared.rs

```diff
@@ -261,7 +261,7 @@ pub struct Directories {
 impl Directories {
     pub fn create(
         mount_finder: &MountFinder,
-        metadata: &CargoMetadata,
+        metadata: &mut CargoMetadata,
```

Comment:
to ensure that further path references are correct, we specify this as `&mut`

Reply:
Not sure I like this design: we should probably either:
1. Factor this out into 2 functions (one to find the mount path of the target directory and one for `Directories::create`)
2. Rename the function so it's clear it modifies `CargoMetadata`.